x/participationrewards/keeper/rewards_validatorSelection.go

```diff
@@ -17,23 +17,25 @@ import (
 // individually in a callback.
 func (k Keeper) AllocateValidatorSelectionRewards(ctx sdk.Context) {
 	k.icsKeeper.IterateZones(ctx, func(_ int64, zone *icstypes.Zone) (stop bool) {
-		k.Logger(ctx).Info("zones", "chain_id", zone.ChainId, "performance address", zone.PerformanceAddress.Address)
-
-		// obtain zone performance account rewards
-		rewardsQuery := distrtypes.QueryDelegationTotalRewardsRequest{DelegatorAddress: zone.PerformanceAddress.Address}
-		bz := k.cdc.MustMarshal(&rewardsQuery)
-
-		k.IcqKeeper.MakeRequest(
-			ctx,
-			zone.ConnectionId,
-			zone.ChainId,
-			"cosmos.distribution.v1beta1.Query/DelegationTotalRewards",
-			bz,
-			sdk.NewInt(-1),
-			types.ModuleName,
-			ValidatorSelectionRewardsCallbackID,
-			0,
-		)
+		if zone.PerformanceAddress != nil {
+			k.Logger(ctx).Info("zones", "chain_id", zone.ChainId, "performance address", zone.PerformanceAddress.Address)
+
+			// obtain zone performance account rewards
+			rewardsQuery := distrtypes.QueryDelegationTotalRewardsRequest{DelegatorAddress: zone.PerformanceAddress.Address}
+			bz := k.cdc.MustMarshal(&rewardsQuery)
+
+			k.IcqKeeper.MakeRequest(
+				ctx,
+				zone.ConnectionId,
+				zone.ChainId,
+				"cosmos.distribution.v1beta1.Query/DelegationTotalRewards",
+				bz,
+				sdk.NewInt(-1),
+				types.ModuleName,
+				ValidatorSelectionRewardsCallbackID,
+				0,
+			)
+		}
 		return false
 	})
```

Comment:
What should we do for the case where the perf address is `nil`?  Anything to report or handle?

Reply:
i dont think so, if there are no performance delegation we dont have the benchmark on distribution. so we skip it

Reply:
Correct. We can skip the zone if the perf account isn’t set. It’s just a race condition after zone creation, between epoch end and ICA initialisation.